packages/block-editor/src/components/use-moving-animation/index.js

```diff
@@ -125,17 +133,8 @@ function useMovingAnimation(
 			: `translate3d(${ x }px,${ y }px,0)`;
 		ref.current.style.zIndex = ! isSelected || isMoving ? '' : '1';
 
-		if (
-			adjustScrolling &&
-			scrollContainer.current &&
-			! prefersReducedMotion
-		) {
-			const blockRect = ref.current.getBoundingClientRect();
-			const diff = blockRect.top - transform.clientTop;
-
-			if ( diff ) {
-				scrollContainer.current.scrollTop += diff;
-			}
+		if ( preserveScrollPosition ) {
+			preserveScrollPosition();
 		}
```

Comment:
Just noting that using optional chaining, we can now do things like this, if we want to:
```suggestion
		preserveScrollPosition?.();
```

Reply:
@youknowriad What do you prefer? This or always return a function?

Reply:
@ZebulanStanphill I'm getting a linting error:

```
Expected an assignment or function call and instead saw an expression.eslintno-unused-expressions
```

Reply:
I have a small preference for always returning a function but I consider all the three variations as code style preferences that don't deserve to be enforced. So your pick :)

Reply:
I don't have a preference, and the other option creates an error, so let's go for noop.